src/js/node/dgram.ts

```diff
@@ -690,127 +753,133 @@ Socket.prototype.remoteAddress = function () {
 };
 
 Socket.prototype.setBroadcast = function (arg) {
-  throwNotImplemented("setBroadcast", 10381);
-  /*
-  const err = this[kStateSymbol].handle.setBroadcast(arg ? 1 : 0);
-  if (err) {
-    throw new ErrnoException(err, 'setBroadcast');
+  const handle = this[kStateSymbol].handle;
+  if (!handle?.socket) {
+    throw new Error("setBroadcast EBADF");
```

Comment:
We should be using a using errors with codes instead of errors with messages. It sounds like we need a helper for throwing a system error with a syscall name and an errno.